spinta/manifests/xsd/helpers.py

```diff
@@ -0,0 +1,641 @@
+import os
+from pathlib import Path
+from urllib.parse import urlparse
+
+from lxml.etree import _Element
+
+from spinta.components import Context
+from lxml import etree, objectify
+from urllib.request import urlopen
+from pprint import pprint
+
+from spinta.utils.naming import to_property_name, to_model_name, Deduplicator
+
+
+class XSDReader:
+
+    def __init__(self, path, dataset_name):
+        self._path = path
+        self.models = []
+        self._custom_types = {}
+        self._dataset_given_name = dataset_name
+        self._set_dataset_and_resource_info()
+
+
+    # mapping of XSD datatypes to DSA datatypes
+    # XSD datatypes: https://www.w3.org/TR/xmlschema11-2/#built-in-datatypes
+    # DSA datatypes: https://atviriduomenys.readthedocs.io/dsa/duomenu-tipai.html#duomenu-tipai
+    # todo finish mapping and make sure all things are mapped correctly
+    DATATYPES_MAPPING = {
+        "string": "string",
+        "boolean": "boolean",
+        "decimal": "number",
+        "float": "number",
+        "double": "number",
+
+        # Duration reikia mapinti su number arba integer, greičiausiai su integer ir XML duration
+        # reikšmė konvertuoti į integer reikšmę nurodant prepare funkciją, kuri konveruoti duration
+        # į integer, papildomai property.ref stulpelyje reikia nurodyti laiko vienetus:
+        # https://atviriduomenys.readthedocs.io/dsa/vienetai.html#laiko-vienetai
+        # todo add prepare functions
+        "duration": "",
+
+        "dateTime": "datetime",
+        "time": "time",
+        "date": "date",
+        "gYearMonth": "",
+        "gYear": "",
+        "gMonthDay": "",
+        "gDay": "",
+        "gMonth": "",
+        "hexBinary": "",
+        "base64Binary": "",
+        "anyURI": "uri",
+        "QName": "",
+        "NOTATION": "",
+        "normalizedString": "string",
+        "token": "string",
+        "language": "string",
+        "NMTOKEN": "string",
+        "NMTOKENS": "",
+        "Name": "",
+        "NCName": "",
+        "ID": "",
+        "IDREF": "",
+        "IDREFS": "",
+        "ENTITY": "",
+        "ENTITIES": "",
+        "integer": "integer",
+        "nonPositiveInteger": "",
+        "negativeInteger": "",
+        "long": "integer",
+        "int": "integer",
+        "short": "integer",
+        "byte": "integer",
+        "nonNegativeInteger": "integer",
+        "unsignedLong": "integer",
+        "unsignedInt": "integer",
+        "unsignedShort": "integer",
+        "unsignedByte": "integer",
+        "positiveInteger": "integer",
+        "yearMonthDuration": "",
+        "dayTimeDuration": "",
+        "dateTimeStamp": "",
+        "": "",
```

Comment:
Įdomu, ką reiškia šis atvejis? Ir kas atsitinka, jei sumapinus gauname vietoje tipo empty string?

Reply:
įdėjau string